File: mr-python/src/inverted-index/InvertedIndexMapper.py
import sys
import os

for line in sys.stdin:
    # Get the words from the line
    words = line.strip().split()
    # Get file name and offset
    file_name = os.environ.get('map.input.file', None)
    # Get the starting offset of the split
    offset = os.environ.get('map.input.start', None)

    for word in words:
        # Accumulate the length of the word
        offset += len(word)
        print '%s\t%s' % (word, file_name + ':' + offset)

File: inverted-index/src/main/java/InvertedIndexMapper.java
import org.apache.hadoop.io.LongWritable;
import org.apache.hadoop.io.Text;
import org.apache.hadoop.mapreduce.Mapper;
import org.apache.hadoop.mapreduce.lib.input.FileSplit;


/*
 * Sample Data words and words
 * it is what it is
 * what is it
 * it is a banana
 */

public class InvertedIndexMapper extends Mapper<LongWritable, Text, Text, Text> {

    @Override
    public void map(LongWritable key, Text value, Context context)
            throws java.io.IOException, InterruptedException {

        //Get the input file name
        FileSplit fs = ((FileSplit) context.getInputSplit());
        String filePathString = fs.getPath().getName();

        //Split the words separated by blanks
        String[] words = value.toString().split(" ");

        //Why the fuck can't we have a foreach?
        for(String word: words) {
            // Until I find a better way output is word fileName:start
            context.write(new Text(word), new Text(filePathString+":"+
                    (fs.getLength() - fs.getStart())
            ));
        }
    }
}
